src/main.py

```diff
@@ -0,0 +1,51 @@
+"""
+A small streamlit app to demonstrate a Parrot Paraphrasing on T5 Model from Huggingface
+Model on HF: https://huggingface.co/prithivida/parrot_paraphraser_on_T5
+
+This is a project for 2022 ML Course practice
+"""
+import streamlit as st
+from parrot_model import ParrotModel
+
+
+@st.cache(allow_output_mutation=True)
+def get_parrot() -> ParrotModel:
+    """Generates the parrot model and caches it, as it takes up around 10-15 seconds"""
+    return ParrotModel()
+
+
+def main() -> None:
+    """Streamlit app entrypoint"""
+    st.set_page_config(
+        page_title="Cockatoo",
+        page_icon="favicon.ico",
+    )
+
+    parrot = get_parrot()
+
+    sidebar = st.sidebar
+    sidebar.header("Configure cockatoo")
+    sidebar.subheader("Cockatoo parameters")
+
+    parrot.set_seed(int(sidebar.number_input("Seed", 0, 999999, 1)))
+    parrot.adequacy_threshold = sidebar.slider("Adequacy", 0.0, 1.0, 0.8)
+    parrot.fluency_threshold = sidebar.slider("Fluency", 0.0, 1.0, 0.8)
+    parrot.amount = int(sidebar.slider("Maximum amount", 1, 9, 3))
+    parrot.do_diverse = sidebar.checkbox("Do Diverse?")
```

Comment:
That doesn't look like a good way to interact between objects. To use class someone need to separately change values (moreover, there are no getters/setters to ensure invariance) and then call the method to generate.

I suggest you pass these values in `para_phrase` method together with text input.

Reply:
Done. Now para_phrase takes 4 more positional arguments, which were implemented as a fields in `ParrotModel`